base_layer/wallet/src/text_message_service/service.rs

```diff
@@ -218,452 +369,114 @@ impl TextMessageService {
         self.screen_name = Some(screen_name);
     }
 
-    pub fn add_contact(&mut self, contact: Contact, conn: &SqliteConnection) -> Result<(), TextMessageError> {
-        let found_contact = Contact::find(&contact.pub_key, conn);
+    pub async fn add_contact(&mut self, contact: Contact) -> Result<(), TextMessageError> {
+        let conn1 = self
+            .database_connection_pool
+            .clone()
+            .get()
+            .map_err(|_| TextMessageError::R2d2Error)?;
+        let contact_clone = contact.clone();
+        let found_contact = poll_fn(move |_| blocking(|| Contact::find(&contact_clone.pub_key, &conn1))).await?;
         if let Ok(c) = found_contact {
             if c.pub_key == contact.pub_key {
                 return Err(TextMessageError::ContactAlreadyExists);
             }
         }
 
-        contact.commit(&conn)?;
-
-        // Send ping to the contact so that if they are online they will flush all outstanding messages for this node
-        let mut oms = self.oms.clone().ok_or(TextMessageError::OMSNotInitialized)?;
-        self.runtime.block_on(oms.send_message(
-            BroadcastStrategy::DirectPublicKey(contact.pub_key.clone()),
-            MessageFlags::empty(),
-            TariMessageType::new(NetMessage::PingPong),
-            PingPong::Ping,
-        ))?;
+        let conn2 = self
+            .database_connection_pool
+            .clone()
+            .get()
+            .map_err(|_| TextMessageError::R2d2Error)?;
+        let contact_clone2 = contact.clone();
+        poll_fn(move |_| blocking(|| contact_clone2.commit(&conn2))).await??;
 
         trace!(
             target: LOG_TARGET,
             "Contact Added: Screen name: {:?} - Pub-key: {} - Address: {:?}",
-            contact.screen_name.clone(),
-            contact.pub_key.clone(),
-            contact.address.clone()
+            contact.screen_name,
+            contact.pub_key,
+            contact.address,
         );
+
+        // Send ping to the contact so that if they are online they will flush all outstanding messages for this node
+        // TODO This was removed as it created random lock ups in the tests, Once the new Future based comms Middleware
+        // is in put this back
+        //        let _ = self
```

Comment:
Perhaps need to look if liveness is silently panicking and is no longer around to respond. It's a very simple service, so keen to see what you've been able to find out.

Reply:
I will def track this down in the next PR now that the new middleware stuff is in as part of reintroducing the applications